lib/setupmanagers/qemuhck/qemu_machine.rb

```diff
@@ -374,9 +374,25 @@ def full_replacement_map
                          @define_variables)
     end
 
+    def read_dynamic_device(device)
+      file_full_path = File.realpath("#{DEVICES_JSON_DIR}/#{device}.rb")
+      require file_full_path
+
+      raise InvalidConfigFile, "#{device} does not exist" unless QemuHCK::Devices.respond_to?(device)
+
+      device_obj = QemuHCK::Devices.public_send(device, @logger)
+
+      raise InvalidConfigFile, "#{device} does not exist" unless device_obj
```

Comment:
This check is unnecessary. Add `nil` checks only where you know the value can be `nil`; otherwise we will need add `nil` checks everywhere.